app/controllers/assignment_invitations_controller.rb

```diff
@@ -144,6 +140,10 @@ def current_invitation
     @current_invitation ||= AssignmentInvitation.find_by!(key: params[:id])
   end
 
+  def current_invitation_status
+    current_invitation.status(current_user)
```

Comment:
Can we memoize this? In it's current state, we're invoking `current_invitation#status` every time this method gets called.

```ruby
@current_invitation_status ||= current_invitation.status(current_user)
```

Or, if a current invitation status can be nil:

```ruby
return @current_invitation_status if defined?(@current_invitation_status)

@current_invitation_status = current_invitation.status(current_user)
```

Reply:
Thanks for catching this.
`current_invitation` shouldn't be `nil` since its a 1:1 dependency on the `Assignment` model. Will go ahead with the former approach ☝️